cumulus_library/study_parser.py

```diff
@@ -665,29 +668,62 @@ def _execute_build_queries(
     # Database exporting functions
 
     def export_study(
-        self, db: databases.DatabaseBackend, data_path: pathlib.Path
+        self,
+        db: databases.DatabaseBackend,
+        schema_name: str,
+        data_path: pathlib.Path,
+        archive: bool,
     ) -> list:
         """Exports csvs/parquet extracts of tables listed in export_list
-
         :param db: A database backend
-        :returns: list of executed queries (for unit testing only)
+        :param schema_name: the schema/database to target
+        :param data_path: the path to the place on disk to save data
+        :param archive: If true, get all study data and zip with timestamp
+        :returns: a list of queries, (only for unit tests)
         """
+        table_list = []
+        path = pathlib.Path(f"{data_path}/{self.get_study_prefix()}/")
         self.reset_counts_exports()
+        if archive:
+            table_query = base_templates.get_show_tables(
+                schema_name, f"{self.get_study_prefix()}__"
+            )
+            result = db.cursor().execute(table_query).fetchall()
+            for row in result:
+                table_list.append(row[0])
+        else:
+            table_list = self.get_export_table_list()
         queries = []
         for table in track(
-            self.get_export_table_list(),
-            description=f"Exporting {self.get_study_prefix()} counts...",
+            table_list,
+            description=f"Exporting {self.get_study_prefix()} data...",
         ):
-            query = f"select * from {table}"
+            query = f"SELECT * FROM {table}"
             dataframe = db.execute_as_pandas(query)
-            path = pathlib.Path(f"{data_path}/{self.get_study_prefix()}/")
             path.mkdir(parents=True, exist_ok=True)
             dataframe = dataframe.sort_values(
                 by=list(dataframe.columns), ascending=False, na_position="first"
             )
             dataframe.to_csv(
                 f"{path}/{table}.csv", index=False, quoting=csv.QUOTE_MINIMAL
             )
-            dataframe.to_parquet(f"{path}/{table}.parquet", index=False)
-            queries.append(query)
+            if not archive:
+                dataframe.to_parquet(f"{path}/{table}.parquet", index=False)
```

Comment:
Why not parquet too in this case? Just space reasons?

This method is starting to feel a little awkward. You start with forked logic (which tables) and end with forked logic (how to handle the results). Only the middle is shared. Should the middle be moved to a helper method and you have two different methods here, like:
```
if archive:
    do_archive()
else:
    do_export()

def do_archive():
   get tables
   do_inner_export()
   zip results

def do_export():
    get tables
    do_inner_export
    parquet results
```

Or maybe instead move some of the "gross" specialized code (like "getting all tables" or "zipping up all tables") into helpers, so that this method can focus just on the if/else-ing.

I dunno. No action needed per se, just started feeling like a lot of very different code based on some `if` conditions.

Reply:
re: parquet, my original thought was 'this is for parking data ahead of a paper', since that where this request originated, and since that's all csvs in our current use case, i deferred to that. but there's no reason :not: to parquet, and i guess that enables you to upload an older set of data to the aggregator, so maybe it's worth doing - and that would decrease some of the branching logic. so lemme do that and then i'll see how that feels w.r.t. breaking things out.

I have a nagging doubt about some of the  infrastructure in this file -  specifically, since its starting to get large, so I think that biases me towards trying to get everything in one place rather than making more functions. 

This might be a bad idea. I feel like at :some: point this might need to get broken out into a set of functions per arg inheriting from a base, but i'm reluctant to go that far at the moment, which might be driving some other less sustainable architecture choices.

Reply:
moved zipping out to utils - that and the parquet cleanup help.